src/object/cli_ec.c

```diff
@@ -245,9 +254,46 @@ ec_update_params(struct ec_params *params, daos_iod_t *iod, d_sg_list_t *sgl,
 	if (nrecx == NULL)
 		return -DER_NOMEM;
 	params->niod.iod_recxs = nrecx;
-	for (i = 0; i < iod->iod_nr; i++)
-		params->niod.iod_recxs[params->niod.iod_nr++] =
-			iod->iod_recxs[i];
+	for (i = 0; i < iod->iod_nr; i++) {
+		if (iod->iod_recxs[i].rx_nr * iod->iod_size > len * k) {
+			/* can't have more than one stripe in a recx entry */
+			uint64_t rem = iod->iod_recxs[i].rx_nr * iod->iod_size;
+			uint64_t start = iod->iod_recxs[i].rx_idx;
+
+			while (rem) {
+				if (rem <= len * k) {
+					params->
+					niod.iod_recxs[params->niod.iod_nr].
+					rx_nr = rem/iod->iod_size;
```

Comment:
define a local variable like this will make the code be easier to read?
 daos_iod_t  *niod = &params->niod;

Also, I can't see what the prefixed "n" means from the comment, could you explain?